airflow/hooks/webhdfs_hook.py

```diff
@@ -0,0 +1,81 @@
+from airflow.hooks.base_hook import BaseHook
+from airflow.configuration import conf
+import logging
+
+from hdfs import InsecureClient, HdfsError
+from airflow.utils import AirflowException
+
+
+class WebHDFSHookException(AirflowException):
+    pass
+
+
+class WebHDFSHook(BaseHook):
+    """
+    Interact with HDFS. This class is a wrapper around the hdfscli library.
+    """
+    def __init__(self, webhdfs_conn_id='webhdfs_default'):
+        self.webhdfs_conn_id = webhdfs_conn_id
+
+    def get_conn(self):
+        """
+        Returns a hdfscli InsecureClient object.
+        """
+        nn_connections = self.get_connections(self.webhdfs_conn_id)
+        for nn in nn_connections:
+            try:
+                logging.debug('Trying namenode {}'.format(nn.host))
+                client = InsecureClient('http://{nn.host}:{nn.port}'.format(nn=nn))
```

Comment:
Don’t worry. We will test it and create a PR if required. Thanks for having it in!

If I get around to get a minisecure cluster running on travis we can test it there. Ground work is there but will still take some effort.

> Op 6 nov. 2015, om 20:54 heeft Arthur Wiedmer <notifications@github.com> het volgende geschreven:
> 
> In airflow/hooks/webhdfs_hook.py <https://github.com/airbnb/airflow/pull/604#discussion_r44180230>:
> 
> > +class WebHDFSHook(BaseHook):
> > +    """
> > +    Interact with HDFS. This class is a wrapper around the hdfscli library.
> > +    """
> > +    def __init__(self, webhdfs_conn_id='webhdfs_default'):
> > +        self.webhdfs_conn_id = webhdfs_conn_id
> > +
> > +    def get_conn(self):
> > +        """
> > +        Returns a hdfscli InsecureClient object.
> > +        """
> > +        nn_connections = self.get_connections(self.webhdfs_conn_id)
> > +        for nn in nn_connections:
> > +            try:
> > +                logging.debug('Trying namenode {}'.format(nn.host))
> > +                client = InsecureClient('http://{nn.host}:{nn.port}'.format(nn=nn))
> Alright, I added a micron of support for this. I have no ability to test this other that it does not interfere with using the InsecureClient on our setup. Let me know if you get the chance to test it. The API provided by the extra seems fairly basic <http://hdfscli.readthedocs.org/en/latest/api.html#module-hdfs.ext.kerberos>, I do not know if it has all you need.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/airbnb/airflow/pull/604/files#r44180230>.
>